### PR TITLE
IPO tool - Upgrade

### PR DESCRIPTION
**Pull request by gabrielctn**
**Changes**
+ Removed GALAXY_SLOT variable for nSlaves, use default instead
+ Removed samplebyclass option, instead 5% but at least 10 files are used (peak picking step) and files are chosen according to factor combinations associated to them ( retcor + grouping step)
+ Added the requirement of a sample metadata file (tabular, W4M style: isa2W4M) to indicate the experimental design factors for retention time correction and grouping parameters optimization, by filling the phenoData in xcmsSet object with factor classes found in investigation file.
+ Updated help  

@sneumann could you have a look ?

### CODE REVIEW

File: tools/phenomenal/ms/ipo/macros.xml [hunk 1]
```diff
@@ -21,9 +21,6 @@
         sh -c "exit \$return"
     </token>
 
-    <token name="@COMMAND_NSLAVES@">
-        nSlaves \${GALAXY_SLOTS:-1}
```
Comment (pcm32) [inline tools/phenomenal/ms/ipo/macros.xml]: I wouldn't remove this, I would just set a sensible default after the :- part that makes sense with the assignment of the tool in the resource bins (tiny, small, medium, large, xlarge).

Comment (gabrielctn) [reply tools/phenomenal/ms/ipo/macros.xml]: Ok so setting `nSlaves \${GALAXY_SLOTS:-4}` works with a destination set to large only, so I think it is only a matter of material, I will put back GALAXY_SLOTS then, thanks for the remark.
I think also that last time I didn't changed the resources allocated to minikube at start (`minikube start --cpus 4 -- memory 8192`), which was certainly the cause of the problem.

Comment (pcm32) [reply tools/phenomenal/ms/ipo/macros.xml]: Great! In the future what we should aim for is to make those slaves processes distributable in Galaxy in some way. Current issue you have is that all these slaves are unfortunately running on the same machine. Maybe we could generate a fake collection of files signalling the different slaves process that we want, and then having another galaxy box to run that part receiving that collection of files plus the R object with the data serialised in a file, and then another box for merging results.

Comment (pcm32) [reply tools/phenomenal/ms/ipo/macros.xml]: We would need to deal somehow with the iterative nature of the process though...

Comment (gabrielctn) [reply tools/phenomenal/ms/ipo/macros.xml]: That would be awesome indeed, I don't know if galaxy can (already) manage this kind of recursive process

**Comment by gabrielctn** [comment]
Oh and so yes it runs fine on minikube.

**Pull request merged**